src/mpi/request/request_impl.c

```diff
@@ -97,196 +92,151 @@ int MPIR_Cancel(MPIR_Request * request_ptr)
   fn_exit:
     return mpi_errno;
   fn_fail:
-
     goto fn_exit;
 }
 
-/* -- Grequest -- */
-
-/* Any internal routines can go here.  Make them static if possible.  If they
-   are used by both the MPI and PMPI versions, use PMPI_LOCAL instead of
-   static; this macro expands into "static" if weak symbols are supported and
-   into nothing otherwise. */
-void MPIR_Grequest_complete(MPIR_Request * request_ptr)
-{
-    MPIR_Request_complete(request_ptr);
-}
-
-/* preallocated grequest classes */
-#ifndef MPIR_GREQ_CLASS_PREALLOC
-#define MPIR_GREQ_CLASS_PREALLOC 2
-#endif
-
-static MPIR_Grequest_class MPIR_Grequest_class_direct[MPIR_GREQ_CLASS_PREALLOC];
-
-static MPIR_Object_alloc_t MPIR_Grequest_class_mem = { 0, 0, 0, 0, MPIR_GREQ_CLASS,
-    sizeof(MPIR_Grequest_class),
-    MPIR_Grequest_class_direct,
-    MPIR_GREQ_CLASS_PREALLOC,
-    NULL
-};
-
-/* We jump through some minor hoops to manage the list of classes ourselves and
- * only register a single finalizer to avoid hitting limitations in the current
- * finalizer code.  If the finalizer implementation is ever revisited this code
- * is a good candidate for registering one callback per greq class and trimming
- * some of this logic. */
-static int MPIR_Grequest_registered_finalizer = 0;
-static MPIR_Grequest_class *MPIR_Grequest_class_list = NULL;
-
-/* Any internal routines can go here.  Make them static if possible.  If they
-   are used by both the MPI and PMPI versions, use PMPI_LOCAL instead of
-   static; this macro expands into "static" if weak symbols are supported and
-   into nothing otherwise. */
-static int MPIR_Grequest_free_classes_on_finalize(void *extra_data ATTRIBUTE((unused)))
+int MPIR_Request_free_impl(MPIR_Request * request_ptr)
 {
```

Comment:
Can you please separate grequest commit also into a separate commit and please explain the reason?

Reply:
done.